hq/app/logic/SnykDisplay.scala

```diff
@@ -0,0 +1,74 @@
+package logic
+
+import com.fasterxml.jackson.core.JsonParseException
+import play.api.libs.json._
+import utils.attempt.{Attempt, FailedAttempt, Failure}
+
+import scala.concurrent.ExecutionContext
+import model._
+import model.Serializers._
+
+object SnykDisplay {
+
+  def getOrganisation(s: String, requiredOrganisation: SnykOrganisationName): Attempt[SnykOrganisation] = {
+    val organisation = for {
+      orglist <- parseJsonToOrganisationList(s)
+      org <- orglist.find(_.name == requiredOrganisation.value)
+    } yield org
+    val error = parseJsonToError(s)
+    val f = Failure(s"Unable to find organisation from $s", s"Could not read Snyk response (${error.error})", 502, None, None)
+    Attempt.fromOption(organisation, FailedAttempt(f))
+  }
+
+  private def parseJsonToError(s: String): SnykError = try {
+    Json.parse(s).asOpt[SnykError].getOrElse(SnykError(s))
+  } catch {
+    case _: JsonParseException => SnykError(s)
+  }
+
+  private def parseJsonToOrganisationList(s: String): Option[List[SnykOrganisation]] = try {
```

Comment:
If this returned an Attempt you'd have the opportunity to give a more precise error message?

Reply:
(which might eliminate the need to parse the JSON twice)

Reply:
It parses it once in an attempt to get the data from a correct response, then once to get an error message from a failed response.  If all fails, it will just use the entire response body.